scripts/build/gen_kobject_list.py

```diff
@@ -138,7 +138,12 @@ def kobject_to_enum(kobj):
 net_sockets = [ ]
 
 def subsystem_to_enum(subsys):
-    return "K_OBJ_DRIVER_" + subsys[:-11].upper()
+    if subsys.endswith("_driver_api"):
+        name = subsys[:-11]
+    else:
+        name = subsys
+
+    return "K_OBJ_DRIVER_" + name.upper()
```

Comment:
Not a fan of this, but should we only allow `__subsystem` declarations with a `_driver_api` suffix? IMO it obfuscates an implementation detail.

Reply:
`__subsystem` requires `_driver_api` suffix to work. How userspace works with device drivers is a mess, ref https://github.com/zephyrproject-rtos/zephyr/pull/71090 https://github.com/zephyrproject-rtos/zephyr/pull/71402

Reply:
There's indeed a "violation" in tree already.
https://github.com/zephyrproject-rtos/zephyr/blob/792fd57d96327dee3d7477479489ebd9fc42f907/include/zephyr/mgmt/ec_host_cmd/backend.h#L106-L109